### ISSUE TITLE
Cannot open links from vscode-fhs under KDE Plasma 5

### ISSUE THREAD

**Issue by linabeee** [NixOS/nixpkgs#145354]
### Describe the bug
Inside KDE Plasma 5, opening links from Visual Studio Code installed with vscode-fhs, like the one to log in to Settings Sync, opens a prompt to trust the web browser, which then fails with an error because the .desktop file is not writable, since it's a symlink to the nix store.

### Steps To Reproduce
1. Install and open vscode-fhs
2. Open any link (easiest way is to go to cog menu at the bottom of the side panel, click Turn on Settings Sync and open either the Microsoft or GitHub links)

### Expected behavior
Links open in the default browser configured in KDE System Settings.

### Screenshots

![trust](https://user-images.githubusercontent.com/36717206/141117176-8dfd5149-bfa5-4881-afbd-9e30519bc913.png)

![error](https://user-images.githubusercontent.com/36717206/141117239-763701b3-040d-4f75-85bf-e3ab2c087d3d.png)

### Additional context
This is nixos-unstable, and I use chrome (😬)

### Notify maintainers

vscode: @eadwu @Synthetica9 @maxeaubrey Plasma: @ttuegel @nyanloutre 

### Metadata

```console
~/nixpkgs/pkgs/desktops/plasma-5 $ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.77, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11pre328987.c935f5e0add"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: vscode-fhs
# a list of nixos modules affected by the problem
module: plasma5
```

**Comment by schlichtanders** [comment]
Also happens to me on nixos 21.11 with vscode and zettlr

**Comment by schlichtanders** [comment]
also happens in marktext

**Comment by schlichtanders** [comment]
@linabeee do you have any success in fixing this?

**Comment by schlichtanders** [comment]
@eadwu @Synthetica9 @maxeaubrey (vscode)
@ttuegel @nyanloutre (Plasma)

can somebody help?

**Comment by schlichtanders** [comment]
Today is the day that this bug is limiting the functionality of a vscode extension

concretely vscode-revealjs is not able to export a reveal.js presentation to PDF because it cannot open chrome.

**Comment by schlichtanders** [comment]
@fabaff Can you take a look and in case ping someone who may know what is going on?

**Comment by schlichtanders** [comment]
dear nixpkgs support, it would be nice to have someone responsible for this kind of problem joining this issue.

As you can see I've already pinged many people who might feel responsible, but no one does, so a fallback is needed.

**Comment by eadwu** [comment]
I'd look into why KIO Client needs the file to be writable. 

I do not experience this issue as I do not use KDE, VSCode opens links fine for me, obeying my xdg settings.

**Comment by schlichtanders** [comment]
@lina-bh any progress?

**Comment by hqurve** [comment]
Is this related to this https://github.com/NixOS/nixpkgs/issues/160923?

**Comment by schlichtanders** [comment]
> Is this related to this #160923?

yes that looks like the very same issue. Thank you for the pointer

**Comment by CnTeng** [comment]
I add pkgs.xdg-utils, and then it works. I find the method form #137871

**Comment by DaKingof** [comment]
Still, an issue even though I tried all these fixed.

Is this why I can't install plugins without using the nix repo versions as well(i.e. pkgs.vscode-extensions.matklad.rust-analyzer)? I thought the point of using this FHS  version was so we didn't have to use the versions in the repo.

**Comment by Thiago-Assis-T** [comment]
Hey, I installed the package xdg-utils through home manager and it worked, it looked like (in my case) that the xdg-open hooks that opens browser and apps was not working, and looking into it and searching I found out that other non-electron apps that tried to open the browser and when they were supposed to be opened from the browser, were not working... I found it in a nixos discourse thing, but I lost the link. It would be worth trying guys
